src/main/java/impl/pair/PairImpl.java

```diff
@@ -0,0 +1,34 @@
+package impl.pair;
+
+import api.pair.Pair;
+
+
+public class PairImpl<T, E> implements Pair<T, E> {
+    private final T first;
+    private final E second;
+
+    PairImpl(T first, E second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    @Override
+    public T getFirst() {
+        return this.first;
+    }
+
+    @Override
+    public E getSecond() {
+        return this.second;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        return toString().equals(obj.toString());
```

Comment:
What if obj is null? What if obj is another class? 
It is a bad practice to check equality using strings representation. (think about why this is so)